mono/metadata/icall.c

```diff
@@ -4075,7 +4082,10 @@ ves_icall_RuntimeType_GetMethodsByName_native (MonoReflectionTypeHandle ref_type
 		return g_ptr_array_new ();
 	}
 
-	return mono_class_get_methods_by_name (klass, mname, bflags, ignore_case, FALSE, error);
+	if (mlisttype == MLISTTYPE_All)
```

Comment:
Pass mlisttype to `mono_class_get_methods_by_name` and handle it there

Reply:
I got worried when a change to the signature required a change in debugger-agent.c line 4299 (and a few other lines).  

If I just change the signature and not the debugger-agent, then FALSE is going to default to 0, which is (MLISTTYPE.All).

My worry was how to validate it didn't break anything :-).

Thoughts?

Reply:
We have a test suite for that, although I am not sure this case is covered the logic there is using same BindingFlags so it should match